packages/core/src/validate-at-property.ts

```diff
@@ -0,0 +1,66 @@
+import type * as postcss from 'postcss';
+import { stripQuotation } from './utils';
+
+const UNIVERSAL_SYNTAX_DEFINITION = '*';
+
+interface AtPropertyValidationResponse {
+    valid: boolean;
+    message?: string;
+    remove?: boolean;
+}
+
+export const atPropertyValidationWarnings = {
+    MISSING_REQUIRED_DESCRIPTOR(descriptorName: string) {
+        return `@property rules require a "${descriptorName}" descriptor`;
+    },
+    MISSING_REQUIRED_INITIAL_VALUE_DESCRIPTOR() {
+        return '@property "initial-value" descriptor is optional only if the "syntax" is the universal syntax definition, otherwise the descriptor is required';
+    },
+};
+
+export function validateAtProperty(atRule: postcss.AtRule): AtPropertyValidationResponse {
+    const atPropertyValues: Map<string, string> = new Map();
+
+    if (!atRule.nodes?.length) {
+        return {
+            valid: true,
+            remove: true,
+        };
+    }
+
+    for (const node of atRule.nodes) {
+        if (node.type !== 'decl') {
```

Comment:
why not warn on unknown content? we should check if other declaration/rules might break validity

Reply:
The check for nested at-rules/rules is good, but I think the most helpful warning here is for unknown declaration. In the case someone would mistakenly types one of the intended declaration wrong (e.g. `initialvalue`)